src/x509.c

```diff
@@ -372,6 +372,7 @@ int meth_extensions(lua_State* L)
         break;
       }
     }
+    sk_GENERAL_NAME_free(values);
```

Comment:
This is not a safe location for this, all the above `push` functions may throw.
You need to put `values` somewhere that `__gc` can take care of it.

Reply:
Okay, this goes over my head both w.r.t. lua and this library. Somebody else will have to do that, or give me some guidance. I have for example no idea which places ``__gc`` (whatever that is?) can take care of (I suspect it’s the GC of Lua, but why do you call it ``__gc``?).

Reply:
@daurnimator, what kind of error are you thinking?
I mean, depending on the error, could it be worse or invalidate the memory leak? (out of memory?)

Reply:
@brunoos I don't understand your question here.
any of the pushing functions may fail when low on memory; you should safeguard this by adding a `__gc` method to clean them up.

For an example of this see my lua-psl library: https://github.com/daurnimator/lua-psl/blob/061b6cefb0a299de15085a20a1e51e51ad252dc8/psl/psl.c#L287-L296 https://github.com/daurnimator/lua-psl/blob/061b6cefb0a299de15085a20a1e51e51ad252dc8/psl/psl.c#L259

Similarly over in luaossl we have a helper function: https://github.com/wahern/luaossl/blob/c5bf3d5272e38b312ff1e3334c0dc3cbdd461caa/src/openssl.c#L651

Reply:
I agree with @mwild1, there are several points in luasec that need to be modified if we follow this line.